volatility3/framework/plugins/windows/windowstations.py

```diff
@@ -0,0 +1,236 @@
+
+        try:
+            symbol_filename = next(
+                filename
+                for version_check, filename in WindowStations._win_version_file_map
+                if version_check(context=context, symbol_table=symbol_table)
+            )
+        except StopIteration:
+            raise NotImplementedError("This version of Windows is not supported!")
+
+        vollog.debug(f"Using GUI table {symbol_filename}")
+
+        return intermed.IntermediateSymbolTable.create(
+            context,
+            config_path,
+            os.path.join("windows", "gui"),
+            symbol_filename,
+            class_types=gui.class_types,
+            native_types=native_types,
+            table_mapping=table_mapping,
+        )
+
+    @classmethod
+    def get_session_map(
+        cls,
+        context: interfaces.context.ContextInterface,
+        module_name: str,
+        gui_table_name: str,
+    ) -> Dict[int, interfaces.context.ModuleInterface]:
+        """
+        Walks each session layer and returns a dictionary that
+        maps session identifiers to a module in the session's layer
+        """
+        session_map = modules.Modules.get_session_layers_map(context, module_name)
+
+        for session_id, session_layer in session_map.items():
+            session_module = context.module(
+                gui_table_name, layer_name=session_layer, offset=0
+            )
+            session_map[session_id] = session_module
+
+        return session_map
+
+    @classmethod
+    def scan_gui_object(
+        cls,
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        kernel_module_name: str,
+        object_tag: bytes,
+        object_type: str,
+    ) -> Generator[interfaces.objects.ObjectInterface, None, None]:
+        """
+        An API that generically scans for GUI (win32*.sys) objects allocated in the pools (which is nearly all of them)
+
+        This function scans within the kernel space for the tags and then uses `get_session_map` to instantiate objects
+        in their correct session address space.
+
+        Args:
+            context:
+            config_path:
+            kernel_module_name:
+            object_tag: The 4 byte pool header tag to search for
+            object_type: The data structure of the GUI object within the pool
+        """
+
+        kernel = context.modules[kernel_module_name]
+
+        gui_table_name = cls.create_gui_table(
+            context, kernel.symbol_table_name, config_path
+        )
+
+        constraints = poolscanner.PoolScanner.gui_poolscanner_constraints(
+            gui_table_name, [object_tag]
+        )
+
+        session_map = cls.get_session_map(context, kernel_module_name, gui_table_name)
+
+        for result in poolscanner.PoolScanner.generate_pool_scan_extended(
+            context,
+            kernel.layer_name,
+            kernel.symbol_table_name,
+            gui_table_name,
+            constraints,
```

Comment:
```suggestion
            context=context,
            kernel_layer_name=kernel.layer_name,
            kernel_symbol_table=kernel.symbol_table_name,
            object_symbol_table=gui_table_name,
            constraints=constraints,
```
Ideally, this would be updated to `kernel_symbol_table_name` and `object_symbol_table_name` although actually, for the extended one, we probably want `kernel_module_name` instead of the symbol and layer names taken separately...

Reply:
See above, it will after this PR. There will be a PR just to cleanly overhaul the poolscanner APIs and callers.

Reply:
I'd still at least apply the suggested changes to use keyword args (and also, please tag the specific ticket that this'll get fixed by, so we can track it).